warehouse/manage/views.py

```diff
@@ -1443,66 +1455,149 @@ def manage_project_roles(project, request, _form_class=CreateRoleForm):
                 f"address and cannot be added as a {role_name} for project",
                 queue="error",
             )
-        else:
-            request.db.add(
-                Role(user=user, project=project, role_name=form.role_name.data)
+        elif (
+            user_invite
+            and user_invite.invite_status == RoleInvitationStatus.Pending.value
+        ):
+            request.session.flash(
+                f"User '{username}' already has an active invite. "
+                f"Please try again later.",
+                queue="error",
```

Comment:
There's an edge case and a 5 minute window here where the invite might be expired, but it's still "Pending" in the database. I think in the case that the user already has an invite, we need to check whether it's expired as well, and manually handle the status change.